datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/util/PatchResolverUtils.java

```diff
@@ -0,0 +1,353 @@
+package com.linkedin.datahub.graphql.resolvers.mutate.util;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.linkedin.common.urn.Urn;
+import com.linkedin.common.urn.UrnUtils;
+import com.linkedin.data.template.StringMap;
+import com.linkedin.datahub.graphql.QueryContext;
+import com.linkedin.datahub.graphql.generated.ArrayPrimaryKeyInput;
+import com.linkedin.datahub.graphql.generated.PatchOperationInput;
+import com.linkedin.datahub.graphql.generated.StringMapEntryInput;
+import com.linkedin.datahub.graphql.generated.SystemMetadataInput;
+import com.linkedin.metadata.aspect.patch.GenericJsonPatch;
+import com.linkedin.metadata.utils.GenericRecordUtils;
+import com.linkedin.metadata.utils.SystemMetadataUtils;
+import com.linkedin.mxe.GenericAspect;
+import com.linkedin.mxe.SystemMetadata;
+import jakarta.json.Json;
+import jakarta.json.JsonPatch;
+import jakarta.json.JsonPatchBuilder;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class PatchResolverUtils {
+
+  // Entity types that support auto-generated URNs (simple string URNs)
+  public static final Set<String> AUTO_GENERATE_ALLOWED_ENTITY_TYPES =
+      Set.of(
+          "glossaryTerm",
+          "glossaryNode",
+          "container",
+          "notebook",
+          "domain",
+          "dataProduct",
+          "ownershipType");
+
+  /**
+   * Resolves entity URN from input - either provided URN or auto-generated for supported entity
+   * types
+   */
+  @Nonnull
+  public static Urn resolveEntityUrn(@Nonnull String urn, @Nullable String entityType)
```

Comment:
This is not valid you shouldn't be patching an urn or aspect into existence. Patch applies to existing entity aspects.